pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/UnnecessaryModifierRule.java

```diff
@@ -264,6 +266,14 @@ public Object visit(ASTConstructorDeclaration node, Object data) {
         return data;
     }
 
+    @Override
+    public Object visit(ASTRecordDeclaration node, Object data) {
+        if (node.isStatic()) {
```

Comment:
It occurs to me that records are also implicitly final. Could you add a branch here and a test case?